# Python User Define Function Practice Activity - 2 

Asst. Prof. Syed Faisal Ali              $\;\;\;\;\;\;$              Programming Fundamentals - FALL 2019 $\;\;\;\;\;\;$ Software Engineering                     $\;\;\;\;\;\;$                   Dated: 27 Nov 2019

Question 1: Create a function to find the following:
If the base of triangle is 3 cm long and its equilateral triangle and the radius of circle is 1.5 cm then find the area of triangle shaded. 


![proj](q1.jpg)

In [1]:
def area(b,r):
    h=b*r/2
    return h
area(3,1.5)

2.25

Question 2: Create a function which can read a dictionary of your family members such as 5 members. 
1 Abbu, 1 Ammi, 2 Brothers 1 Sister. Now feed this data in dictionary in terms of name and relations.
The UDF will ask findrelation() in this you will enter Brother it will return the names of two brothers you have inserted. In case if the relation is not found it will return “Sorry the relation doesn’t exist in your family.” 


In [2]:
def findrelation(b):
    dict = {'father':'Abdul Haq' , 'brother':'shoaib, zubair' , 'sister':'eman' , 'mother':'hajra'}
    a = dict[b]
    return a
findrelation('brother')

'shoaib, zubair'

Question 3: Create a function to find the following:
If the base of triangle is 5 cm long and its equilateral triangle and the radius of circle is 2.25 cm then find the area of triangle shaded. 


![proj](q3.jpg)

In [3]:
import math
def func(b):
    h=b**2*((math.sqrt(3))/4)
    return(h)
func(5)

10.825317547305483

Question 4:
Create a function that takes a list of random numbers from users and add only those which are even. If all the numbers are odd it will return sorry no even number found.


In [12]:
def A(l):
    s=0
    for i in l:
        if i%2==0:
            print(i)
            s = s+i
            print("sum of even no:",s)
        else:
            print("Sorry no even found.")
print(A([1,2,3,4,5,6,7,8,9,10]))

Sorry no even found.
2
sum of even no: 2
Sorry no even found.
4
sum of even no: 6
Sorry no even found.
6
sum of even no: 12
Sorry no even found.
8
sum of even no: 20
Sorry no even found.
10
sum of even no: 30
None


Question 5:
Write a function which can take a list of numbers and it will return sorted list.


In [11]:
def list_(l):
    return sorted(l)
print (list_(['9','1','8','-2']))

['-2', '1', '8', '9']


Question 6:
Write a function that will take the radius and return the perimeter and area of circle with 5% increment.


In [31]:
from math import pi
def area_circle(r):
    area=pi*(r**2)
    a = area+(area*0.05)
    print("Perimeter:",2*pi*r)
    print("Area of Circle:",a)
    return " "
print(area_circle(6))


from math import pi
def radius(r):
    area=pi*r**2
    perimeter=2*pi*r
    return(area,perimeter)
radius(6)

Perimeter: 37.69911184307752
Area of Circle: 118.75220230569418
 


(113.09733552923255, 37.69911184307752)

Question 7:
Write a function that will take the strings as argument and return number of vowels and consonants.


In [50]:
def check(str):
    vlist=['a','e','i','e','o','u','A','E','I','O','U']
    vcount=0
    ccount=0
    for i in str.lower():
        if i in vlist:
            vcount+=1
        else:
            ccount+=1
    print("Vowels are", vcount)
    print("consonants are", ccount)
check('My Name Is Muhammad Zeeshan')

Vowels are 9
consonants are 18


Question 8:
Write a function that will take length and breadth for a rectangle and return perimeter and area of rectangle with 8% increment.


In [52]:
from math import pi
def rec(l,b):
    area=l*b
    a=area+(area*0.08)
    print("area of rectangle:",a)
    print("perimeter:",2*(l+b))
rec(4,9)  

area of rectangle: 38.88
perimeter: 26



Question 9:
Write a function that can take the numbers in strings. From string find which number is even and which one is odd. Save them in two different lists and generate the result.


In [55]:
string = "123456789"
e = 0
o = 0
el = []
ol = []
for i in string:
    if int(i)%2==0:
        e += 1
        el.append(i)
    else:
        o += 1
        ol.append(i)
print('Numbers of Even :',e)
print('Numbers of odd:',o)
print("List of even:",el)
print("List of odd:",ol)

Numbers of Even : 4
Numbers of odd: 5
List of even: ['2', '4', '6', '8']
List of odd: ['1', '3', '5', '7', '9']


Question 10:
Write a function which will take the string from the user and return how many alphabets have been used in it and which alphabets are missing.


In [62]:
string = "muhammad zeeshan"
n = 0
a = 'abcdefghijklmnopqrstuvwxyz'
for i in string:
    if i in a:
        a = a.replace(i,"")
        n += 1
print(n)
print(a)

9
bcfgijklopqrtvwxy


Question 11:
Write a function that will take verbs in words and return a list of verbs with continuous tense by adding (ing) at the end of each verb.


In [65]:
def verb(a):
    if a[-1] == 'e':
        a = a.replace(a[-1],"")
        print("verb",a+"ing")
    else:
        print("verb",a+"ing")
verb('work')

verb working


Question 12:
Make a function which can take two radius of circles and find the areas of it and subtract smaller one from larger one and tell the remaining area of circle.


In [66]:
from math import pi
def cc(r,rr):
    a = pi*(r**2)
    aa = pi*(rr**2)
    if a > aa:
        fa = (aa - a)
    else:
        fa = (a - aa)
    return abs(fa)
print(cc(0,13))

530.929158456675


Question 13:
Write a function that will take a string and calculate number of Upper case letters and lower case letters.


In [69]:
def ul(s):
    nu = 0
    nl = 0
    l = 'abcdefghijklmnopqrstuvwxyz'
    u = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    for i in s:
        if i in u:
            nu = nu+1
        if i in l:
            nl = nl+1
    print(nl,nu)
print(ul("Zeeshan"))

6 1
None


Question 14:
Write a function which will take length and breadth of two rectangles. Subtract the smaller rectangle from the larger rectangle and return the area left behind.


In [78]:
def cc(l1,b1,l2,b2):
    a = l1*b1
    aa = l2*b2
    if a > aa:
        fa = (aa - a)
    else:
        fa = (a - aa)
    return abs(fa)
print(cc(5,7,9,13))

82


Question 15:
Create a function that can add the fractions in series such as 1 to 8 = 1/8+1/7+1/6+1/5 …… ½ and return the result in fraction not in decimal.


In [79]:
from fractions import Fraction
i = 1 
f = 8
ff = 0
for ii in range(i,f):
    ff = ff + Fraction(1,ii)
print(ff)

363/140


Question 16:
Write a function which will take height and base for a triangle and 


Question 17:
Write a function which will take a list of fruits names. The function will return how many alphabets are repetitive in the names of fruits and how many are unique letters.


In [85]:
f = 'Guava'
F = ""
nn = 0
for i in f:
    n = f.count(i)
    if n > 1:
        nn = nn + 1
print('rep',nn)

rep 2


Question 18:
Write a function that can take square length and radius of circle. Find the area of both and subtract the smallest shape from largest one and return the remaining shape area.


In [81]:
def sc(l,r):
    sa = l*l
    ca = 3.142*(r**2)
    print(sa)
    print(ca)
    if sa > ca:
        fa = (ca - sa)
    else:
        fa = (sa - ca)
    return abs(fa) 
sc(4,7)

16
153.958


137.958